rudderlabs/rudder-sdk-kotlin
# feat(android): add locale os screen timezone plugins

## Description


In this PR, we've added the following context plugins:

- [x] Locale
- [x] OS
- [x] Screen
- [x] Timezone

## Type of change


## Implementation Details


We have added following context fields:
- LocaleInfoPlugin
- OSInfoPlugin
- ScreenInfoPlugin
- TimezoneInfoPlugin

## Checklist


## How to test?


Run the app, make events and observe the `context` filed in the payload. You can also compare the results with v1 SDK results.

Payload:
```
{
    "anonymousId": "<anonymous-id>",
    "channel": "mobile",
    "context": {
        "app": {
            "build": "100",
            "name": "Rudder-Android-Libs",
            "namespace": "com.rudderstack.android.sampleapp",
            "version": "0.1.0"
        },
        "library": {
            "name": "com.rudderstack.android.sdk",
            "version": "1.0.0"
        },
        "locale": "en-US",
        "os": {
            "name": "Android",
            "version": "14"
        },
        "screen": {
            "density": 560,
            "height": 2808,
            "width": 1440
        },
        "timezone": "Asia/Kolkata"
    },
    "event": "Application Opened",
    "integrations": {
        "All": true
    },
    "messageId": "203e9b16-329d-4d02-85f2-483d8e566100",
    "originalTimestamp": "2024-10-03T14:07:54.803Z",
    "properties": {
        "from_background": false,
        "version": "0.1.0"
    },
    "receivedAt": "2024-10-03T14:07:58.316Z",
    "request_ip": "49.37.35.25",
    "rudderId": "69d539cd-377e-403c-becb-1cb36e7ddc6a",
    "sentAt": "2024-10-03T14:07:56.205Z",
    "type": "track"
}
```

## Breaking Changes


## Maintainers Checklist


## Screenshots (if applicable)

<img width="316" alt="image" src="https://github.com/user-attachments/assets/0c9dfbd6-4ca5-4de2-8004-b0e214d0d760">


## Additional Context
